scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/goals/Procedure.java

```diff
@@ -64,24 +65,24 @@ public void run(
         problem.getRealExternalProfiles()
     );
 
-    final var editablePlan = new InMemoryEditablePlan(
+    final var editAdapter = new SchedulerPlanEditAdapter(
         missionModel,
         idGenerator,
         planAdapter,
         simulationFacade,
         lookupActivityType::apply
     );
 
+    final var editablePlan = new DefaultEditablePlanDriver(editAdapter);
+
     procedureMapper.deserialize(SerializedValue.of(this.args)).run(editablePlan);
 
-    if (!editablePlan.getUncommittedChanges().isEmpty()) {
+    if (editablePlan.isDirty()) {
       throw new IllegalStateException("procedural goal %s had changes that were not committed or rolled back".formatted(jarPath.getFileName()));
     }
     for (final var edit : editablePlan.getTotalDiff()) {
       if (edit instanceof Edit.Create c) {
         newActivities.add(toSchedulingActivity(c.getDirective(), lookupActivityType::apply, true));
-      } else {
-        throw new IllegalStateException("Unexpected value: " + edit);
```

Comment:
Why was this removed instead of changing the `else` to an `else if (!(edit instanceof Edit.Delete))`? Additionally, is there any value in having a procedure track updated and deleted activities as well as added ones?

Reply:
I can put the error back, no reason not to.

I don't think the procedure could do anything with the record of deleted activities, yet. We only record the new activities so the `source_scheduling_goal` field can be set in the database, which obviously doesn't exist for deleted activities.